src/config_file.c

```diff
@@ -503,17 +545,24 @@ RaucSlot *find_slot_by_device(GHashTable *slots, const gchar *device)
 {
 	GHashTableIter iter;
 	RaucSlot *slot;
+	g_autofree MountableObj *obj = normalize_mountable_object(device);
+
+	if (!obj) {
+		return NULL;
+	}
 
 	g_return_val_if_fail(slots, NULL);
 	g_return_val_if_fail(device, NULL);
 
 	g_hash_table_iter_init(&iter, slots);
 	while (g_hash_table_iter_next(&iter, NULL, (gpointer*) &slot)) {
```

Comment:
For compatibility reasons maybe we should not break the prior behavior of this function.

Is there any disadvantage if we simply do the former string comparison as a fallback here? This would assure that we match more cases but not lose others. Checking if the device that was listed in the config actually exists, etc. should anyway be handled somewhere else.
This would also remove the need for (`find_config_slot_by_name`, despite it might still be useful as convenience fo tests, etc.